node/silkworm/downloader/internals/body_retrieval.cpp

```diff
@@ -20,15 +20,15 @@
 
 namespace silkworm {
 
-BodyRetrieval::BodyRetrieval(Db::ReadOnlyAccess db_access) : db_tx_{db_access.start_ro_tx()} {}
+BodyRetrieval::BodyRetrieval(db::ROAccess db_access) : db_tx_{db_access.start_ro_tx()} {}
 
 std::vector<BlockBody> BodyRetrieval::recover(std::vector<Hash> request) {
     std::vector<BlockBody> response;
     size_t bytes = 0;
     for (size_t i = 0; i <= request.size(); ++i) {
         Hash& hash = request[i];
         BlockBody body;
-        if (!db_tx_.read_body(hash, body)) {
+        if (!db::read_body(db_tx_, hash, body)) {
             continue;
         }
```

Comment:
Maybe `for(Hash& hash : request)` is more concise.

Reply:
ok

Reply:
No, there is "i" usage